File: python/346. Moving Average from Data Stream/solution.py
class MovingAverage:

    def __init__(self, size: int):
        self.size = size
        self.total = 0
        self.queue = []

    def next(self, val: int) -> float:
        self.total += val
        self.queue.append(val)
        if len(self.queue) > self.size:
            removed = self.queue.pop(0)
            self.total -= removed

        return self.total / len(self.queue)


# Your MovingAverage object will be instantiated and called as such:
# obj = MovingAverage(size)
# param_1 = obj.next(val)

File: java/346. Moving Average from Data Stream/Solution.java
class MovingAverage {
    double total;
    Queue<Integer> queue;
    int size;

    public MovingAverage(int size) {
        this.size = size;
        this.total = 0;
        this.queue = new LinkedList<Integer>();
    }

    public double next(int val) {
        total += val;
        queue.add(val);
        if (queue.size() > size) {
            total -= queue.poll();
        }
        return total / queue.size();
    }
}

/**
 * Your MovingAverage object will be instantiated and called as such:
 * MovingAverage obj = new MovingAverage(size);
 * double param_1 = obj.next(val);
 */
 

class MovingAverage {
    double total;
    ArrayList<Integer> queue;
    int size;

    public MovingAverage(int size) {
        this.size = size;
        this.total = 0;
        this.queue = new ArrayList<Integer>();
    }

    public double next(int val) {
        total += val;
        queue.add(val);
        if (queue.size() > size) {
            total -= queue.get(0);
            queue.remove(0);
        }
        return total / queue.size();
    }
}